client/extensions/woocommerce/app/products/index.js

```diff
@@ -6,42 +6,35 @@ import { bindActionCreators } from 'redux';
 import classNames from 'classnames';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
+import { trim } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import ActionHeader from 'woocommerce/components/action-header';
 import Button from 'components/button';
-import EmptyContent from 'components/empty-content';
-import { fetchProducts } from 'woocommerce/state/sites/products/actions';
+import { fetchProducts, fetchProductSearchResults, clearProductSearch } from 'woocommerce/state/sites/products/actions';
 import { getLink } from 'woocommerce/lib/nav-utils';
-import { getTotalProducts, areProductsLoaded } from 'woocommerce/state/sites/products/selectors';
-import { getProductListCurrentPage, getProductListProducts, getProductListRequestedPage } from 'woocommerce/state/ui/products/selectors';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
 import Main from 'components/main';
-import Pagination from 'my-sites/stats/pagination';
-import ProductsListTable from './products-list-table';
+import ProductsList from './products-list';
+import ProductsListSearchResults from './products-list-search-results';
 import SidebarNavigation from 'my-sites/sidebar-navigation';
+import SearchCard from 'components/search-card';
 
 class Products extends Component {
 	static propTypes = {
 		site: PropTypes.shape( {
 			slug: PropTypes.string,
 		} ),
-		products: PropTypes.oneOfType( [
-			PropTypes.array,
-			PropTypes.bool,
-		] ),
-		currentPage: PropTypes.number,
-		currentPageLoaded: PropTypes.bool,
-		requestedPage: PropTypes.number,
-		requestedPageLoaded: PropTypes.bool,
-		totalProducts: PropTypes.number,
-		fetchProducts: PropTypes.func,
+		fetchProducts: PropTypes.func.isRequired,
+		fetchProductSearchResults: PropTypes.func.isRequired,
+		clearProductSearch: PropTypes.func.isRequired,
 	};
 
-	// Total number of results per page (the API returns 10)
-	perPage = 10;
+	state = {
```

Comment:
should our components be stateful?  Could this be pushed to redux?  Not needed for now but something to think about post launch